### DISCUSSION TITLE
VPN

### DISCUSSION THREAD

**Discussion by shawnthompson** [htpcBeginner/docker-traefik#112] [Show and tell]
Moving this issue #35 to a discussion on VPN and network setups. Feel free to discussion your solutions for your network setup. I'm going to try @robflate's setup and see if that works.

**Comment by shawnthompson** [comment]
So I have I'm running:
```
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
``` 
which is dedicated as a media server running Treafik version 2.

What if I setup the whole system to run through a VPN instead of adding it to this compose file?

Pros and cons?

**Comment by marrrc** [comment]
I am trying to run the example setup from @robflate in #35 with autoheal and gluetun. seems to work fine, until a container restart is imminent because gluetun was e.g. restarted. autoheal correctly picks up the depending containers, but all i get is this:

` 2021-03-14T16:45:07.358884369Z 14-03-2021 16:45:07 Container /jackett (7824c9fd65dc) found to be unhealthy - Restarting container now with 10s timeout
autoheal               | 2021-03-14T16:45:11.000926052Z 14-03-2021 16:45:07 Restarting container 7824c9fd65dc failed`

so autoheal manages to run its  `restart_container` method, however after stopping the container doesn't come back up and stays in stopped state. any env variable maybe that is missing for me on the socket-proxy regarding API use?

edit: seems that restarting an unhealthy container via docker API gets me a `curl: (22) The requested URL returned error: 500 Internal Server Error`, not sure why though.
